### PR TITLE
Use a single actions service

### PR DESCRIPTION
**Pull request by itaiad200**

### CODE REVIEW

File: pkg/actions/store.go [hunk 1]
```diff
@@ -0,0 +1,297 @@
+package actions
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/treeverse/lakefs/pkg/db"
```
Comment (N-o-Z) [inline pkg/actions/store.go:8]: Redundant new lines

File: pkg/actions/store.go [hunk 2]
```diff
@@ -0,0 +1,297 @@
+package actions
```
Comment (N-o-Z) [inline pkg/actions/store.go:1]: I would still split implementations into 2 files so we can just delete the whole file when we finish the KV migration

File: pkg/actions/service.go [hunk 3]
```diff
@@ -37,7 +36,8 @@ var (
 )
 
 type KVService struct {
```
Comment (N-o-Z) [inline pkg/actions/service.go:38]: "Service"?

**Review by N-o-Z — CHANGES_REQUESTED** [review]
Very small refactoring... XD
Added comments - seems like tests are failing as well

File: pkg/actions/service.go (hunk 3 above)
Comment (itaiad200) [reply pkg/actions/service.go]: It's the interface, changed the name to something else.

File: cmd/lakefs/cmd/run.go [hunk 4]
```diff
@@ -180,7 +180,9 @@ var runCmd = &cobra.Command{
 		bufferedCollector.CollectMetadata(metadata)
 
 		var multipartsTracker multiparts.Tracker
```
Comment (N-o-Z) [inline cmd/lakefs/cmd/run.go:182]: Suggest to join the var declarations

File: cmd/lakefs/cmd/run.go [hunk 5]
```diff
@@ -191,26 +193,26 @@ var runCmd = &cobra.Command{
...
 			multipartsTracker = multiparts.NewTracker(storeMessage)
-			actionsService = actions.NewKVService(
-				ctx,
-				storeMessage,
-				catalog.NewActionsSource(c),
-				catalog.NewActionsOutputWriter(c.BlockAdapter),
-				bufferedCollector,
-				cfg.GetActionsEnabled(),
-			)
+
```
Comment (N-o-Z) [inline cmd/lakefs/cmd/run.go:196]: Redundant new lines?

File: cmd/lakefs/cmd/run.go [hunk 6]
```diff
@@ -191,26 +193,26 @@ var runCmd = &cobra.Command{
...
 			multipartsTracker = multiparts.NewDBTracker(dbPool)
-			actionsService = actions.NewDBService(
-				ctx,
-				dbPool,
-				catalog.NewActionsSource(c),
-				catalog.NewActionsOutputWriter(c.BlockAdapter),
-				bufferedCollector,
-				cfg.GetActionsEnabled(),
-			)
+
```
Comment (N-o-Z) [inline cmd/lakefs/cmd/run.go:201]: same

File: pkg/loadtest/local_load_test.go [hunk 7]
```diff
@@ -9,6 +9,8 @@ import (
 	"testing"
 	"time"
 
+	"github.com/treeverse/lakefs/pkg/kv"
+
```
Comment (N-o-Z) [inline pkg/loadtest/local_load_test.go:13]: Newline

File: pkg/actions/id.go [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+import (
+	"time"
+
+	"github.com/rs/xid"
+
+	gonanoid "github.com/matoous/go-nanoid/v2"
+	"github.com/treeverse/lakefs/pkg/graveler"
+)
+
+type IDGenerator interface {
```
Comment (N-o-Z) [inline pkg/actions/id.go:12]: Do we want to move the NewHookRunID functionality also under this interface?

File: pkg/actions/id.go [hunk 9]
```diff
@@ -0,0 +1,33 @@
+package actions
+
+import (
+	"time"
+
+	"github.com/rs/xid"
+
```
Comment (N-o-Z) [inline pkg/actions/id.go:7]: newline

**Review by N-o-Z — CHANGES_REQUESTED** [review]
Looks really great, I added some more comments and suggestions

File: cmd/lakefs/cmd/run.go (hunk 5 above)
Comment (itaiad200) [reply cmd/lakefs/cmd/run.go]: Done.

File: cmd/lakefs/cmd/run.go (hunk 6 above)
Comment (itaiad200) [reply cmd/lakefs/cmd/run.go]: Done.

File: pkg/actions/id.go (hunk 8 above)
Comment (itaiad200) [reply pkg/actions/id.go:11]: I think not because it's the same for both implementations.

File: pkg/loadtest/local_load_test.go (hunk 7 above)
Comment (itaiad200) [reply pkg/loadtest/local_load_test.go]: Done.

File: pkg/actions/service.go [hunk 10]
```diff
@@ -36,8 +35,12 @@ var (
 	ErrNilValue = errors.New("nil value")
 )
 
-type KVService struct {
-	Store    kv.StoreMessage
+// MultiStorageService is an implementation of actions.Service that saves
+// the run data to the blockstore and to the actions.Store (which is a
+// fancy name for a DB - kv style or postgres directly)
+type MultiStorageService struct {
```
[the remark this answers is not in the document]
Comment (itaiad200) [reply pkg/actions/service.go]: I hate naming.

**Review by N-o-Z — APPROVED** [review]
Looks good :ok_hand:
Thanks for addressing all the comments

**Pull request merged**